core/src/main/java/com/amplifyframework/api/graphql/GraphQLResponse.java

```diff
@@ -110,24 +114,79 @@ public int hashCode() {
      * See https://graphql.github.io/graphql-spec/June2018/#sec-Response-Format
      */
     public static final class Error {
+        // Description of the error
         private final String message;
 
+        // list of locations describing the syntax element
+        private final List<GraphQLLocation> locations;
+
+        // Details the key path of the response field with error.
+        private final List<GraphQLPathSegment> path;
+
+        // Additional error map, reserved for implementors to use however they see fit.
+        private final Map<String, Object> extensions;
+
         /**
          * Constructs error response in accordance with GraphQL specs.
          * @param message error message
+         * @param locations list of locations describing the syntax element
+         * @param path The key path of the response field with error.
+         * @param extensions additional error map, reserved for implementors.
          */
-        public Error(@NonNull String message) {
+        public Error(@NonNull String message,
+                     @Nullable List<GraphQLLocation> locations,
+                     @Nullable List<GraphQLPathSegment> path,
+                     @Nullable Map<String, Object> extensions) {
```

Comment:
(TLDR Ignore this, no action needed.)

------

I just realized how we could do this 100% to spec. It ends up being way overkill, and I still do think we should stick what you have, here.

It would be possible for us to model the extensions as a native GraphQL entity. To do so, we would need Java models for all of the various [types defined in the GraphQL spec](https://graphql.org/learn/schema/).

iOS uses the type `JSONValue`, here. While GraphQL and JSON share some similarities, the type definitions are _not_ the same. As a simple example, even the scalar definitions are slightly different (ref: [GraphQL](https://graphql.org/learn/schema/#scalar-types), [JSON](https://tools.ietf.org/html/rfc7159#section-3).) The `String`-`Object` `Map` in Java is a _looser_ specification of typing than the GraphQL spec.

Part of the scope of work we have taken up with Amplify is to create a lightweight GraphQL client. The AppSync SDK leverages an [Apollo](https://github.com/apollographql/apollo-android) distribution for this purpose. While we building our GraphQL client, we have identified needed functionalities by breadth-first discovery. But in instances like this, we sometimes uncover considerable depth needed to fully model all of the parts of the spec.

I hadn't checked until now. But for what its worth, [Apollo _also does use_ a `String`-`Object` `Map` for this model, too](https://github.com/apollographql/apollo-android/blob/master/apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/Error.kt#L20).

Reply:
Thanks for the thoughts.  I think this would be a small improvement over what I have now, but for now, I think Map<String, Object> works pretty well.  

It would be nice to actually parse extensions as a template type though.  I think this could actually be done by creating a new template type R (response) which wraps a <T extends Model> and an extension type, E.   This would require updating all methods in ApiBehavior though, and I'm not sure if we have the bandwidth to make that change right now.